### PR TITLE
upgrade react-toast-notifications version (with hooks)

### LINKED ISSUE #1534: upgrade `react-toast-notifications`
@jossmac has updated his [`react-toast-notifications`](https://www.npmjs.com/package/react-toast-notifications) to version 2.2.4. I found it has `useToasts` hooks, which may be useful and help remove wrapping with `withToastManager`

```js
const { addToast, removeToast, toastStack } = useToasts();
```

### PR DESCRIPTION
**Pull request by gautamsi**
closes #1534 
* upgrade react-toast-notifications
* use useToasts hook when possible.

### CODE REVIEW

File: packages/app-admin-ui/client/pages/Item/index.js [hunk 1]
```diff
@@ -373,8 +378,9 @@ const ItemNotFound = ({ adminPath, errorMessage, list }) => (
   </PageError>
 );
 
-const ItemPage = ({ list, itemId, adminPath, getListByKey, toastManager }) => {
+const ItemPage = ({ list, itemId, adminPath, getListByKey }) => {
   const itemQuery = list.getItemQuery(itemId);
+  const { addToast } = useToasts();
```
Comment (jossmac) [inline packages/app-admin-ui/client/pages/Item/index.js]: Might as well pass on all the functionality here? It's being deconstructed later on [line 459](https://github.com/keystonejs/keystone-5/pull/1544/files#diff-18fd6ba6fe6ea61ebfb3250f37792127R459).

```suggestion
  const toastManager = useToasts();
```

File: packages/app-admin-ui/client/pages/Item/index.js [hunk 2]
```diff
@@ -450,7 +456,7 @@ const ItemPage = ({ list, itemId, adminPath, getListByKey, toastManager }) => {
                         onUpdate={() =>
                           refetch().then(refetchedData => deserializeItem(list, refetchedData.data))
                         }
-                        toastManager={toastManager}
+                        toastManager={{ addToast }}
```
Comment (jossmac) [inline packages/app-admin-ui/client/pages/Item/index.js]: Regarding my [earlier comment](https://github.com/keystonejs/keystone-5/pull/1544/files#r318331292):

```suggestion
                        toastManager={toastManager}
```

**Comment by jossmac** [comment]
Actually, don't worry about my comments. We can come back to it later when we've migrated away from class components.

**Comment by gautamsi** [comment]
@jossmac if moving away from class component is a goal (however distant 😄)  I can try help on this. I did not touch so much in this PR to avoid any delays.

**Pull request merged**